src/unix/fs.c

```diff
@@ -150,8 +146,17 @@ static ssize_t uv__fs_fdatasync(uv_fs_t* req) {
 
 static ssize_t uv__fs_fsync(uv_fs_t* req) {
 #if defined(__APPLE__)
-  /* See the comment in uv__fs_fdatasync. */
-  return fcntl(req->file, F_FULLFSYNC);
+  /* Apple's fdatasync and fsync explicitly do NOT flush the drive write cache
+   * to the drive platters. This is in contrast to Linux's fdatasync and fsync
+   * which do, according to recent man pages. F_FULLFSYNC is Apple's equivalent
+   * for flushing buffered data to permanent storage. If F_FULLFSYNC is not
+   * supported by the file system we should fall back to fsync(). This is the
+   * same approach taken by sqlite.
+   */
+  int r = fcntl(req->file, F_FULLFSYNC);
```

Comment:
Can you write as:

```c
int r;

r = fcntl(req->file, F_FULLFSYNC);
```

Reply:
Sure, will make the change.

I'm new to C++, can I ask why the variable declaration should be done separately?

Reply:
Because that's the house style: declarations go at the start of the function, assignments in the function body.

(Also, note that libuv is C code, not C++.)

Reply:
Thanks!